Apps/phdigicert/digicert_connector.py

```diff
@@ -0,0 +1,539 @@
+        except requests.exceptions.InvalidURL:
+            error_message = 'Error connecting to server. Invalid URL {}'.format(url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except requests.exceptions.ConnectionError:
+            error_message = 'Error Details: Connection Refused from the Server {}'.format(url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except Exception as e:
+            return RetVal(
+                action_result.set_status(
+                    phantom.APP_ERROR, "Error Connecting to server. Details: {0}".format(str(e))
+                ), resp_json
+            )
+
+        return self._process_response(r, action_result)
+
+    def _handle_test_connectivity(self, param):
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Connecting to endpoint")
+
+        # make rest call
+        ret_val, response = self._make_rest_call("/user/me", action_result)
+
+        if phantom.is_fail(ret_val):
+            # the call to the 3rd party device or service failed, action result should contain all the error details
+            self.save_progress("Test Connectivity Failed.")
+            self.save_progress(action_result.get_message())
+            return action_result.get_status()
+
+        # Return success
+        self.save_progress("Test Connectivity Passed")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_request_cert(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        container_id = self.get_container_id()
+
+        # required params
+        csr = param["csr"]  # vault_id or filename
+        server_platform_id = consts.PLATFORM_MAP[param["server_platform"]]
+        order_validity_years = int(param["order_validity"])
+
+        # optional params
+        cn = param.get("common_name")
+        signature_hash = param.get("signature_hash")
+
+        success, msg, vault_info = ph_rules.vault_info(container_id=container_id)
+
+        if not success:
+            return action_result.set_status(phantom.APP_ERROR, msg)
+
+        filename = ""
+        for vault_item in vault_info:
+            if csr == vault_item["vault_id"] or csr == vault_item["name"]:
+                filename = vault_item["path"]
+                break
+        else:
+            # 'else' on a for loop is called if it IS NOT broken (i.e. break is never called)
+            # in this case that means we did not find a vault item in the list, so return an error
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Could not find the CSR in the file vault"
```

Comment:
Please use same capitalization throughout app (csr or CSR). In the JSON, lowercase is used

Reply:
@alexa-phantom Done with all the said changes.